### ISSUE TITLE
email for new inbox notifications

### ISSUE THREAD

**Issue by cellio** [codidact/qpixel#1329]
meta:277426
https://software.codidact.com/posts/290885

We should allow users to opt in to email notifications for new on-site (inbox) notifications.  This is particularly helpful for "long tail" questions, where you get an answer or a comment months later, long after you've stopped actively monitoring it.

We already have machinery in place to send other kinds of notifications (daily or less often), so we should be able to build on that.  We don't want to send email upon every new notification (too noisy), but maybe we could offer something finer-grained than daily?

**Comment by RokeJulianLockhart** [comment]
> We don't want to send email upon every new notification (too noisy)

@cellio, I disagree. I triage each e-mail separately (as Proton Mail permits), rather than per thread (like GMail and Outlook). I might find use in summaries at times, but there always needs to be the granularity to have one for each notification, if the mail server can handle it.

(If that's infeasible, I hope that an Atom feed exists, so that I can forward each notification that it exposes to my e-mail inbox as separate messages, instead.)

**Comment by trichoplax** [comment]
My ideal solution would involve a user preference for how often to be notified by email. For example, a user could choose between:

- Never receive emails for new inbox notifications
- Receive a single email once a week if there are any new inbox notifications
- Receive a single email each day if there are any new inbox notifications
- Receive an email as soon as there is a new inbox notification, but then no further emails until the user next visits the site
- Receive an email for every new inbox notification, even if the user has not visited the site since the last email

@RokeJulianLockhart would that last one cover what you are requesting?

If the server only sends out emails, for example, once a day, it should still be possible to send individual emails per notification, they just might all arrive at once at a given time each day. I don't know what's possible/practical. These are just ideas.

**Comment by trichoplax** [comment]
In an answer to the Meta post, I suggested that email notifications be opt out so that people who assume there will be an email don't miss answers to their questions (such as [reported on Software Development Meta](https://software.codidact.com/posts/290885)). That currently has 6 upvotes and 2 downvotes. Would a reasonable middle ground be to sign up new users for the lowest frequency emails (so weekly in the example above) so they might miss an answer by a week, but not indefinitely? If this is only applied to new users, and all existing users have to opt in, then the downvoters won't be opted in to something they've expressed opposition to, but all new users will be prevented from unexpectedly missing notifications due to expecting there to be emails.

**Comment by RokeJulianLockhart** [comment]
1. > Receive an email for every new inbox notification, even if the user has not visited the site since the last email.

1. > If the server only sends out emails, for example, once a day, it should still be possible to send individual emails per notification, they just might all arrive at once at a given time each day.

@trichoplax, either are acceptable, if not ideal. Thanks!

> [`#issuecomment-3218420599`](https://github.com/codidact/qpixel/issues/1329#issuecomment-3218420599)

I'd support that, although I can't imagine that SE's default of a digest per day wouldn't be too invasive of a default. That is, insofar as we ensure that they're solely notified of content that they have explicitly interacted with and followed. [^1]

[^1]: [`meta.discourse.org/t/387087`](https://meta.discourse.org/t/receving-periodic-digests-of-non-followed-shouldnt-be-the-default-are-they/387087#:~:text=Receving%20periodic%20digests%20of%20non%2Dfollowed%20shouldn%E2%80%99t%20be%20the%20default.%20Are%20they?)

**Comment by cellio** [comment]
That's a good point @RokeJulianLockhart ; thanks for raising it.  We probably don't want to send messages immediately and individually *by default*, but defaulting to daily, with all notifications new in that time, seems quite reasonable and consistent with many other services.  And allowing much faster notification seems doable to me, down to "send email at the same time you add it to the notification inbox".  (We've made some improvements to our email setup since I filed this issue.)
